### DISCUSSION TITLE
Non-italic subscripts in math mode

### DISCUSSION THREAD

**Discussion by dnlsch** [typst/typst#3132] [Q&A] [answered]
I would like to write non-italic indices in math mode. The subscript text can also be a word, i.e. more than one character long. How can I set globally that all subscripts in math mode should be non-italic?

**Comment by emilyyyylime** [comment, marked as answer]
If they're a word, you can simply use e.g. `x_"offset"` and it would not be italic. If you want single letters to also be displayed upright when shown in subscripts, you can use this snippet:
```typ
#show math.attach: it => {
  import math: *
  if it.has("b") and it.b.func() != upright[].func() and it.b.has("text") and it.b.text.len() == 1 {
    let args = it.fields()
    let _ = args.remove("base")
    let _ = args.remove("b")
    attach(it.base, b: upright(it.b), ..args)
  } else {
    it
  }
}
```

**Reply by dnlsch** [reply]
Thanks @emilyyyylime , it works!
